airbyte-scheduler/scheduler-persistence/src/main/java/io/airbyte/scheduler/persistence/DefaultJobPersistence.java

```diff
@@ -409,22 +408,19 @@ private static RecordMapper<Record, SyncStats> getSyncStatsRecordMapper() {
   }
 
   private static RecordMapper<Record, NormalizationSummary> getNormalizationSummaryRecordMapper(final Long attemptId) {
+    return record -> new NormalizationSummary().withStartTime(record.get(NORMALIZATION_SUMMARIES.START_TIME).toInstant().toEpochMilli())
+        .withEndTime(record.get(NORMALIZATION_SUMMARIES.END_TIME).toInstant().toEpochMilli())
+        .withFailures(record.get(NORMALIZATION_SUMMARIES.FAILURES, String.class) == null ? null : deserializeFailureReasons(record, attemptId));
+  }
+
+  private static List<FailureReason> deserializeFailureReasons(final Record record, final Long attemptId) {
     final ObjectMapper mapper = new ObjectMapper();
-    return record -> {
-      try {
-        return new NormalizationSummary().withStartTime(record.get(NORMALIZATION_SUMMARIES.START_TIME).toInstant().toEpochMilli())
-            .withEndTime(record.get(NORMALIZATION_SUMMARIES.END_TIME).toInstant().toEpochMilli())
-            .withFailures(record.get(NORMALIZATION_SUMMARIES.FAILURES, String.class) == null ? null
-                : Lists.newArrayList(mapper.readValue(String.valueOf(record.get(NORMALIZATION_SUMMARIES.FAILURES)), FailureReason[].class)));
-      } catch (final JsonProcessingException e) {
-        // There was an error deserializing the FailureReasons into a list of FailureReason Objects,
-        // so we are returning null Failures
-        LOGGER.error("There was an error deserializing NormalizationSummary Failures for attempt id " + attemptId);
-        return new NormalizationSummary().withStartTime(record.get(NORMALIZATION_SUMMARIES.START_TIME).toInstant().toEpochMilli())
-            .withEndTime(record.get(NORMALIZATION_SUMMARIES.END_TIME).toInstant().toEpochMilli())
-            .withFailures(null);
-      }
-    };
+    try {
+      return List.of(mapper.readValue(String.valueOf(record.get(NORMALIZATION_SUMMARIES.FAILURES)), FailureReason[].class));
+    } catch (final JsonProcessingException e) {
```

Comment:
I think we need to rethrow this exception, we are doing it for the IOException, the API will return a 500 error instead of a 200 with a null list.